docs/ADR/0011_smart_contract_storage_list_purge.md

```diff
@@ -0,0 +1,143 @@
+# ADR Template
+
+## Title: Deposit system for the on chain storage & list, purge possible implementation
+
+## Date
+2025-06-05
+Last status update: 2025-09-05
+
+## Status
+- [X] Proposed
+- [ ] Accepted
+- [ ] Rejected
+- [ ] Deprecated
+- [ ] Superseded by ADR #X
+
+### Implementation Status
+- [X] Planned
+- [ ] In Development
+- [ ] Implemented
+- [ ] Verified
+- [ ] Discontinued
+
+## People
+- **Decision-makers**: Vas, Alisher, Andrew
+- **Consulted**: KP
+- **Informed**:
+  - Sviat
+  - Max
+
+## Context
+
+The ADR tries to answer on two questions. Please see sections `Question 1` and `Question 2`.
+
+### Question 1:
+How the fee logic for the storing on chain items, and running operations on top of it?
+Answer on this: the PolkaDot logic respects an idea that everething is going to be fee'ed and every stored item should be deposited, see more at [2] and [3]. For instance:
+- Alice stores two contracts and two random value.
+- Means Alice has been spent (freezed) tokens, i.e. deposite, for storing these data [4][2].
+- To return deposit'ed tokens back: Alice would have to removed stored contracts and random values.
+
+This logic allows to protect the system from the case 'store everething and forever w/o any regulation and pinalties', ex: Eth. stores everething and forever that a huge issue of Eth, since storage size is growing forever and _NOT_ regulated.
```

Comment:
```suggestion
This logic allows to protect the system from the case 'store everything and forever w/o any regulation and penalties', ex: Ethereum. stores everything and forever that a huge issue of Ethereum, since storage size is growing forever and _NOT_ regulated.
```